File: C++/set_match.cpp
// no. 645
#include <map>
#include <vector>
using namespace std;

class Solution {
public:
  vector<int> findErrorNums(vector<int> &nums) {
    map<int, int> counter;
    vector<int> ans = {0, 0};
    int n = nums.size();
    for (int i : nums) {
      counter[i]++;
    }
    for (int i = 1; i < n + 1; i++) {
      map<int, int>::iterator it = counter.find(i);
      if (it->second == 2) {
        ans[0] = i;
      } else if (it == counter.end()) {
        ans[1] = i;
      }
    }
    return ans;
  }
};

File: python/set_match.py
#no. 645

class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        ans = [0, 0]
        n = len(nums)
        counter = {i + 1: 0 for i in range(n)}
        for i in nums:
            counter[i] += 1
        for i in range(n):
            if counter[i + 1] == 2:
                ans[0] = i + 1
            elif counter[i + 1] == 0:
                ans[1] = i + 1
        return ans
